tests/scripts/mbed-build.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/sh
+
+# mbed-build.sh
+#
+# This file is part of mbed TLS (https://tls.mbed.org)
+#
+# Copyright (c) 2017, ARM Limited, All Rights Reserved
+#
+# Purpose
+#
+# To run test builds of the mbed OS module for all supported targets.
+
+set -eu
+
+MBED_APP='mbed-os-example-tls'
+CPUS=$(grep -c ^processor /proc/cpuinfo)
+
+check_tools()
+{
+    for TOOL in "$@"; do
+        if ! `hash "$TOOL" >/dev/null 2>&1`; then
+            echo "$TOOL not found!" >&2
+            exit 1
+        fi
+    done
+}
+
+mbed_build()
+{
+    TARGET=$1
+    TOOLCHAIN=$2
+
+    echo; echo "*** $TARGET (release $TOOLCHAIN) ***"
+    mbed compile -t $TOOLCHAIN -m $TARGET -c -j  $((4*$CPUS))
+
+    echo; echo "*** $TARGET (debug $TOOLCHAIN) ***"
+    mbed compile -t $TOOLCHAIN -m $TARGET --profile mbed-os/tools/profiles/debug.json -c -j  $((4*$CPUS))
+}
+
+# Make sure the tools we need are available.
+check_tools "arm-none-eabi-gcc" "armcc" "mbed"
+
+mbed-cli/create-module.sh
+cd mbed-cli/$MBED_APP
+
+TOOLCHAINS=" ARM GCC_ARM "
+TARGETS=" K64F NUCLEO_F429ZI "
+
+for f in *; do
+	if [ -d $f ]; then
+	    OLD_PWD=$PWD
+		cd $f
```

Comment:
I can see that you are remembering the current working directly so that later you can return to it. However, a better way of doing it could be to use the commands (see http://unix.stackexchange.com/questions/81224/how-to-go-to-the-previous-working-directory-in-terminal):
```
cd $f
# some code...
cd -
```

Alternatively you can use the commands `pushd` and `popd` to achieve the same result http://unix.stackexchange.com/questions/77077/how-do-i-use-pushd-and-popd-commands.